### DISCUSSION TITLE
How to remove unnecessary forced styles and why do i have them in first place?

### DISCUSSION THREAD

**Discussion by sshyam-gupta** [chakra-ui/chakra-ui#2779] [Q&A] [answered]
I am trying to upgrade to chakra-UI v1.x but for some reason, hundreds of styles tags + inline styles are appended to my dom.

![Screen Shot 2020-12-14 at 4 56 12 PM](https://user-images.githubusercontent.com/38554977/102076150-4bbc1580-3e2d-11eb-852f-eadce25d42f2.png)
![Screen Shot 2020-12-14 at 4 56 25 PM](https://user-images.githubusercontent.com/38554977/102076164-4fe83300-3e2d-11eb-894b-78cd1fba2b46.png)

Because of these styles in many places, my UI is breaking.
Example 1:
![Screen Shot 2020-12-14 at 4 57 51 PM](https://user-images.githubusercontent.com/38554977/102076498-cab14e00-3e2d-11eb-9212-7614caf43edd.png)

which breaks my Avatar component:
![Screen Shot 2020-12-14 at 5 00 24 PM](https://user-images.githubusercontent.com/38554977/102076553-e0bf0e80-3e2d-11eb-9c3e-a96b0e76042e.png)

Example 2: Adds unnecessary padding
![Screen Shot 2020-12-14 at 5 04 48 PM](https://user-images.githubusercontent.com/38554977/102077049-b3bf2b80-3e2e-11eb-8f1e-eb7c8c049117.png)
Example snippet:
```jsx
<Slider
  color="green"
   value={progressDuration ? Math.floor(progressDuration.playedSeconds) : 0}
    max={totalDuration}
    onChange={onSliderChange}
    height="4px"
    paddingTop="0.25rem"
    paddingBottom="0"
    min={0}
    onMouseOver={onOpen}
    onMouseOut={onClose}
    zIndex={11}
 >
  <SliderTrack h={1}>
      <SliderFilledTrack h={1} />
    </SliderTrack>
  <SliderThumb size={14} />
</Slider>
```

**Comment by with-heart** [comment, marked as answer]
Hi @sshyam-gupta! 

Chakra components are produced using [`emotion`](https://github.com/emotion-js/emotion). The way `emotion` works is to inject `style` tags containing the styles for each component under randomly-generated classnames that are then attached to the elements produced by each component. Here's what you're seeing in `head`:

- `<style data-emotion="css-global">` tags are for the "global styles" of the Chakra system, which come from our [css reset](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/css-reset/src/css-reset.tsx) and any global styles you set in [`theme.styles.global`](https://chakra-ui.com/docs/features/global-styles).
- `<style data-emotion="css">` tags are for the styles of the individual Chakra components. There might be one or more `style` tags for each component, depending on how many elements the component is composed of.

This works the same way in both Chakra v0.x and v1.x. Compare the `head` in these basic CodeSandbox examples that generate similar output:

- Chakra v0.8.0: https://sdcd9.csb.app/
- Chakra v1.0.3: https://e7q0b.csb.app/

The `<style data-emotion="css">` tags for a component are only injected _if that component is rendered_, so those styles are only "forced" in the sense that those styles must be generated in order for Chakra components to look the way you'd expect them to look.

As for your examples:

1. `AvatarGroup` is used to render multiple stacked avatars: https://chakra-ui.com/docs/data-display/avatar#avatargroup. It appears you're wrapping both `Avatar` and `Text` with it, which will cause them to render incorrectly. If you're trying to render the avatar to the left of the text with space between them, I recommend wrapping with [`HStack`](https://chakra-ui.com/docs/layout/stack#usage) instead.
2. The inline styles are dynamically generated based on the size and orientation of the `Slider` and are injected when the component is rendered: https://github.com/chakra-ui/chakra-ui/blob/770466ae99443d12c214e26032a5ebc704397853/packages/slider/src/use-slider.ts#L337-L354

`padding-top` and `padding-bottom` are dynamically generated based on the height of the thumb and injected to guarantee that there is appropriate space to render the track and thumb and center them vertically within their container.

Hope this helps clear some things up!

**Reply by sshyam-gupta** [reply]
Thank you @with-heart for the detailed explaination.

I got this, but things were working fine before.
While upgrading I am facing these issues.
Most of the places breaking are where I am using `Stack`.

I came across this [issue](https://github.com/chakra-ui/chakra-ui/issues/2578) now and understand the problem here.
maybe we must update the types for the `Stack` and remove margin or m from `SpaceProps`.

**Comment by albertoramires** [comment]
@with-heart what would be the solution to keeps sliders consistent inside tabs? I have two tabs, the slider on the first one (default) renders 8px paddings but the second one renders 0px paddings.
